src/main/java/io/supertokens/cronjobs/bulkimport/ProcessBulkImportUsers.java

```diff
@@ -0,0 +1,597 @@
+
+                if (processedUser != null && isProcessedUserFromSameBulkImportUserEntry(processedUser, user)) {
+                    baseTenantStorage.startTransaction(con2 -> {
+                        baseTenantStorage.deleteBulkImportUser_Transaction(appIdentifier, con2, user.id);
+                        return null;
+                    });
+                    return;
+                }
+            }
+
+            bulkImportProxyStorage.startTransaction(con -> {
+                try {
+                    for (LoginMethod lm : user.loginMethods) {
+                        processUserLoginMethod(appIdentifier, bulkImportProxyStorage, lm);
+                    }
+
+                    createPrimaryUserAndLinkAccounts(main, appIdentifier, bulkImportProxyStorage, user, primaryLM);
+                    createUserIdMapping(main, appIdentifier, user, primaryLM);
+                    verifyEmailForAllLoginMethods(appIdentifier, con, bulkImportProxyStorage, user.loginMethods);
+                    createTotpDevices(main, appIdentifier, bulkImportProxyStorage, user.totpDevices, primaryLM);
+                    createUserMetadata(appIdentifier, bulkImportProxyStorage, user, primaryLM);
+                    createUserRoles(main, appIdentifier, bulkImportProxyStorage, user);
+
+                    // We are updating the primaryUserId in the bulkImportUser entry. This will help us handle the inconsistent transaction commit.
+                    // If this update statement fails then the outer transaction will fail as well and the user will simpl be processed again. No inconsistency will happen in this
+                    // case.
+                    baseTenantStorage.updateBulkImportUserPrimaryUserId(appIdentifier, user.id,
+                            primaryLM.superTokensUserId);
+
+                    // We need to commit the transaction manually because we have overridden that in the proxy storage
+                    // If this fails, the primaryUserId will be updated in the bulkImportUser but it wouldn’t actually exist.
+                    // When processing the user again, we'll check if primaryUserId exists with the same email. In this case the user won't exist, and we'll simply re-process it.
+                    bulkImportProxyStorage.commitTransactionForBulkImportProxyStorage();
+
+                    // NOTE: We need to use the baseTenantStorage as bulkImportProxyStorage could have a different storage than the baseTenantStorage
+                    // If this fails, the primaryUserId will be updated in the bulkImportUser and it would exist in the database.
+                    // When processing the user again, we'll check if primaryUserId exists with the same email. In this case the user will exist, and we'll simply delete the entry.
+                    baseTenantStorage.startTransaction(con2 -> {
+                        baseTenantStorage.deleteBulkImportUser_Transaction(appIdentifier, con2, user.id);
```

Comment:
the plugin interface as a `deleteBulkImportUser`. If we are not using that, remove that from the interface and its implementation

Reply:
I deleted `deleteBulkImportUser_Transaction` instead. We can use `deleteBulkImportUser` in all the places.